src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/nonassociations/NonAssociationsService.kt

```diff
@@ -122,16 +123,30 @@ class NonAssociationsService(
             mappingService.deleteNonAssociation(nonAssociationId)
           }
       }.onSuccess {
-        telemetryClient.trackEvent("nonAssociation-delete-success", telemetryMap)
+        telemetryClient.trackEvent("non-association-delete-success", telemetryMap)
       }.onFailure { e ->
-        telemetryClient.trackEvent("nonAssociation-delete-failed", telemetryMap)
+        telemetryClient.trackEvent("non-association-delete-failed", telemetryMap)
         throw e
       }
     } else {
-      telemetryClient.trackEvent("nonAssociation-delete-ignored", telemetryMap)
+      telemetryClient.trackEvent("non-association-delete-ignored", telemetryMap)
     }
   }
 
+  suspend fun processMerge(event: MergeEvent) {
+    if (event.additionalInformation.reason == "MERGE") {
+      mappingService.mergeNomsNumber(event.additionalInformation.removedNomsNumber, event.additionalInformation.nomsNumber)
+    }
+    telemetryClient.trackEvent(
```

Comment:
seems slightly weird for this to be outside the `if` block?

Reply:
what other reasons could there be?

Reply:
True its the only one atm, i guess i wanted to see something logged regardless

Reply:
ok i put it inside the 'if' as otherwise it is not a 'non-association-merge-success'